### PR TITLE
Make runner task sorting pay special attention to integers which come be...

### PR DESCRIPTION
**Pull request by mrrrgn**
...fore a dash.

### CODE REVIEW

File: runner/lib/graph.py [hunk 1]
```diff
@@ -40,6 +40,26 @@ def _missing_tasks(self):
...
+        if '-' in x and '-' in y:
+            x_pre = x.split('-')[0]
+            y_pre = y.split('-')[0]
+            if x_pre.isdigit() and y_pre.isdigit() and (x_pre != y_pre):
+                x = int(x_pre)
+                y = int(y_pre)
+                # double zeros are a special case: when compared to a single
+                # zero, it should be considered as "less." To make this work
+                # for any number of zeros, we'll convert them into negatives.
+                if (x == 0 and y == 0) and (len(x_pre) > 1 or len(y_pre) > 1):
```
Comment (rail) [inline runner/lib/graph.py]: I think this is very confusing...
00 < 0, but its not the same for 10 vs 010. I believe they will be equal.
I'd rather leave everything as is and just make our task files use consistent number of digits.

**Comment by mrrrgn** [comment]
Fair enough.

File: runner/lib/graph.py (hunk 1 above)
Comment (mrrrgn) [inline runner/lib/graph.py]: We still need a change IMO. I can take out the 00 part, but right now a task prefixed with 100, runs before a task prefixed with 99. Leaving in the split would fix that up.

File: tests/test-graph.py [hunk 2]
```diff
@@ -12,6 +12,23 @@
 )
 
 
+def test_graph_comparator():
+    """
+    We use a custom comparator to ensure that tasks sort the way we want
+    by name (if no config is specified)
+    """
+    assert TaskGraph.task_comparator('99-abc', '100-abc') == -1
```
Comment (mrrrgn) [inline tests/test-graph.py]: @rail This test fails in our current runner for instance.

**Pull request closed without merging**